src/Model/ValueObject/String/Behaviour/HasMaxLength.php

```diff
@@ -4,14 +4,14 @@
 
 namespace CultuurNet\UDB3\Model\ValueObject\String\Behaviour;
 
-use InvalidArgumentException;
+use CultuurNet\UDB3\Model\ValueObject\String\Behaviour\Exception\MaxLengthExceeded;
 
 trait HasMaxLength
 {
-    private function guardTooLong(string $value, int $maxLength, string $path='/'): void
+    private function guardTooLong(string $value, int $maxLength): void
     {
         if (mb_strlen($value) > $maxLength) {
-            throw new InvalidArgumentException(sprintf('Given string should not be longer than %d characters.', $maxLength));
+            throw MaxLengthExceeded::maxLengthExceeded($maxLength);
```

Comment:
I think it is fine to use the constructor here, that would also be more logical to read?

Reply:
I always prefer to make the constructor private for custom exceptions, and put the error message (the string) inside the exception itself. 
This gives a couple of benefits:

- Gives a nice summary of all possible errors this Exception is used
- Gives another developer the reuse the exact the same message, and not have 5 variants of the same error floating in the code
- Forced any developer to make a new type of error explicit

This is also a recommended way i books I have read about this subject.

So, I would prefer to keep it this way if it is oke for you?

Reply:
Fine for me to keep it. But you could also change the constructor to only accept the length and use the fixed string inside the constructor.

I think the examples you are talking about are:

```
EventCannotBeAddedToProduction::becauseItAlreadyBelongsToAnotherProduction
EventCannotBeAddedToProduction::becauseSomeEventsBelongToAnotherProduction
```

In the current implementation, there is only one flavor, and something like `LengthIsInvalid::becauseMaxLengthIsExceeded` could work which reads more logically than `becauseSomeEventsBelongToAnotherProduction`